File: gg/gateways/git/rebase.py
from plumbum import FG

from gg.gateways.git import git_cmd

def git_rebase_interactive(ref: str) -> int:
    """perform an interactive rebase onto the ref"""
    command = git_cmd['rebase', '-i', ref]
    return (command) & FG

def git_rebase_commit_order(ref: str) -> int:
    """perform an interactive rebase onto the ref"""
    command = git_cmd['rebase', ref, '--exec', 'git commit --amend --reset-author --no-edit']
    return (command) & FG

File: go/src/github.com/willhug/gg/git/rebase.go
package git

func (g *Git) RebaseInteractive(ref string) error {
	return g.runner.Run("rebase", "-i", ref)
}

func (g *Git) RebaseAbort() error {
	return g.runner.Run("rebase", "--abort")
}

func (g *Git) RebaseContinue() error {
	return g.runner.Run("rebase", "--continue")
}

// Fix the commit times (can cause github to show the commits out of order)
func (g *Git) RebaseFixCommitTimes(ref string) error {
	return g.runner.Run("rebase", ref, "--exec", "'git commit --amend --reset-author --no-edit'")
}
